File: back/internal/models/event.go
package models

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type Event struct {
	gorm.Model
	ID                uuid.UUID     `gorm:"primaryKey;type:uuid;default:uuid_generate_v4()"`
	Reservations      []Reservation `gorm:"foreignKey:EventID"`
	Messages          []Message     `gorm:"foreignKey:EventID"`
	Rates             []Rate        `gorm:"foreignKey:EventID"`
	Organizers        []Organizer   `gorm:"many2many:organizer_events;references:UserID"`
	Title             string
	Description       string
	Banner            string
	Image             string
	Date              time.Time
	ParticipantNumber *int
	Lat               float32
	Lng               float32
	Location          string
	Place             string
	Tag               string
}

File: mobile/lib/models/event.dart
class Event {
  final String id;
  final String title;
  final String description;
  final String tag;
  final String image;
  final String date;
  final String place;

  Event({
    required this.id,
    required this.title,
    required this.description,
    required this.tag,
    required this.image,
    required this.date,
    required this.place,
  });

  factory Event.fromJson(Map<String, dynamic> json) {
    return Event(
      id : json['id'],
      title: json['title'],
      description: json['description'],
      tag: json['tag'],
      image: json['image'],
      date: json['date'],
      place: json['place'],
    );
  }
}
